I heard @xdustinface encountered installation error on MacOS
```
> root@1.2.10-dev132 bootstrap /xxxxx/chia-blockchain/chia-blockchain-gui
> lerna bootstrap
 
lerna notice cli v4.0.0
lerna info Bootstrapping 6 packages
lerna info Installing external dependencies
lerna ERR! npm install exited 1 in '@chia/api-react'
lerna ERR! npm install stderr:
npm WARN npm npm does not support Node.js v16.14.0
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/
npm ERR! cb.apply is not a function
``` 

I investigated the cause and found possible path to this error.
The facts:
- The error was resolved by removing `~/node_modules`
- `~/node_modules` looks like npm's global installation folder.
- In `install-gui.sh`, I use `n` as a nodejs version switcher and added `bin` folder which contains `n`'s `npm` to `$PATH` variable. And in the script, `n`'s nodejs/npm should be used entirely to build npm dependencies for the GUI.
- The global installation folder for `n`'s `npm` should be `chia-blockchain/.n/lib/node_modules`
- So in theory, `npm WARN npm npm does not support Node.js v16.14.0` should not happen if `n`'s `npm` is actually used.
- But somehow non-`n` version of `npm` is used in `lerna bootstrap` execution.
- I investigated codes below:
  - https://github.com/lerna/lerna/
  - https://github.com/lerna/lerna/blob/main/commands/bootstrap/index.js
  - https://github.com/lerna/lerna/blob/main/utils/npm-install/npm-install.js
  - https://github.com/lerna/lerna/blob/main/utils/get-npm-exec-opts/get-npm-exec-opts.js
  - https://github.com/lerna/lerna/blob/main/core/child-process/index.js
  - https://github.com/sindresorhus/execa#extendenv
  - https://github.com/sindresorhus/execa/blob/main/index.js
- I found one persuasive cause.
  - `lerna bootstrap` internally calls `npm install`
  - To execute `npm install` , `lerna` tries to spawn child process.
  - The child process inherits parent's `process.env` but not `$PATH` shell variable.
  - `install-gui.sh` just adds `bin` containing `n`'s `npm`/`node` to `$PATH` variable, and it never touches `process.env`.
  - I suspect that users who globally installed old `npm` by `npm install -g npm` may encounter this error, since PATH in `process.env` seems to have precedence over `$PATH` shell variable.

--- a/install-gui.sh
+++ b/install-gui.sh
@@ -92,8 +92,8 @@ patch_inconsistent_npm_issue(){
   if [ ! -d "$node_module_dir" ]; then
     mkdir "$node_module_dir"
   fi
-  if [ ! -d "{$node_module_dir}/.bin" ]; then
-    mkdir "{$node_module_dir}/.bin"
+  if [ ! -d "${node_module_dir}/.bin" ]; then
+    mkdir "${node_module_dir}/.bin"
   fi
   if [ -e "${node_module_dir}/.bin/npm" ]; then
     rm -f "${node_module_dir}/.bin/npm"

--- a/install-gui.sh
+++ b/install-gui.sh
@@ -85,6 +85,22 @@ do_install_npm_locally(){
   fi
 }
 
+# Work around for inconsistent `npm` exec path issue
+# https://github.com/Chia-Network/chia-blockchain/pull/10460#issuecomment-1054492495
+patch_inconsistent_npm_issue(){
+  node_module_dir=$1
+  if [ ! -d "$node_module_dir" ]; then
+    mkdir "$node_module_dir"
+  fi
+  if [ ! -d "{$node_module_dir}/.bin" ]; then
+    mkdir "{$node_module_dir}/.bin"
+  fi
+  if [ -e "${node_module_dir}/.bin/npm" ]; then
+    rm -f "${node_module_dir}/.bin/npm"
+  fi
+  ln -s "$(command -v npm)" "${node_module_dir}/.bin/npm"
+}
+
 # Manage npm and other install requirements on an OS specific basis
 if [ "$(uname)" = "Linux" ]; then
   #LINUX=1
@@ -175,9 +191,15 @@ if [ ! "$CI" ]; then
     echo ""
   fi
 
+  # Work around for inconsistent `npm` exec path issue
+  # https://github.com/Chia-Network/chia-blockchain/pull/10460#issuecomment-1054492495
+  patch_inconsistent_npm_issue "../node_modules"
+
   npm ci
   npm audit fix || true
   npm run build
+
+  # Set modified output of `chia version` to version property of GUI's package.json
   python ../installhelper.py
 else
   echo "Skipping node.js in install.sh on MacOS ci."

